openeo/__init__.py

```diff
@@ -17,11 +17,28 @@ class BaseOpenEoException(Exception):
 from openeo.rest.connection import connect, session, Connection
 from openeo.rest.job import BatchJob, RESTJob
 from openeo.internal.graph_building import UDF
-
+import requests
+import warnings
 
 def client_version() -> str:
     try:
         import importlib.metadata
         return importlib.metadata.version("openeo")
     except Exception:
         return __version__
+
+def check_if_latest_version():
+    try:
+        package = 'openeo'
+        response = requests.get(f'https://pypi.org/pypi/{package}/json')
```

Comment:
I think we should put a timeout here of just a couple of seconds so that users are not blocked to do `import openeo` when pypi is slow

Reply:
I also think we should do this check like once a day, or cache the pypi.org response

Reply:
for example, the pip implementation is roughly contained here: https://github.com/pypa/pip/blob/main/src/pip/_internal/self_outdated_check.py 
apparently the check frequency is once a week there

Reply:
I've added the timeout. The pip implementation using the cache would require some more time to be ported over.